File: apps/web/src/types/acMeshLink.ts
export type MeshDataStatus = 'fresh' | 'recent' | 'stale' | 'error' | 'unknown' | 'no_data'
export type MrOnlineStatus = 'online' | 'offline' | 'stale' | 'unknown'

export interface AcMeshLinkSummary {
  site_id: string
  controller_id: string
  controller_name: string
  registered_mrs: number
  online_mrs: number
  offline_mrs: number
  stale_mrs: number
  unknown_mrs: number
  active_links: number
  link_total: number
  unmatched_links: number
  offline_ap_links: number
  updated_at: string
  age_seconds: number | null
  data_status: MeshDataStatus
  source_type: string
  raw_available: boolean
  message: string
}

export interface AcMeshLinkRefreshResponse {
  success: boolean
  task_id: string
  status: string
  already_running: boolean
  message: string
}

export interface AcMeshLinkRecord {
  id: number
  snapshot_id: number
  controller_id: string
  controller_name: string
  mr_id: string
  train_no: string
  car_end: string
  mr_name: string
  mr_mac: string
  mr_device_id: string
  mr_management_ip: string
  mr_online_status: MrOnlineStatus
  peer_ap_id: string
  peer_ap_name: string
  peer_ap_mac: string
  peer_radio: string
  mesh_interface: string
  rssi: number | null
  station: string
  section: string
  mileage: string
  line_side: string
  ap_rx_power: string
  switch_rx_power: string
  last_seen_at: string
  match_method: string
  match_warning: string
  data_status: MeshDataStatus
}

export interface AcMeshLinkPage {
  items: AcMeshLinkRecord[]
  total: number
  page: number
  page_size: number
}

export interface AcMeshMrStatus {
  mr_id: string
  train_no: string
  train_display_name: string
  car_end: string
  mr_name: string
  mr_mac: string
  mr_device_id: string
  management_ip: string
  online_status: MrOnlineStatus
  peer_ap_id: string
  peer_ap_name: string
  peer_ap_mac: string
  mesh_radio: string
  rssi: number | null
  station: string
  section: string
  mileage: string
  line_side: string
  ap_rx_power: string
  switch_rx_power: string
  last_seen_at: string
  match_method: string
  match_warning: string
  data_status: MeshDataStatus
}

export interface AcMeshMrPage {
  items: AcMeshMrStatus[]
  total: number
  page: number
  page_size: number
}

export interface AcMeshMrEvent {
  id: number
  event_time: string
  event_type: string
  status: string
  station: string
  ap_name: string
  rssi: number | null
  car_end: string
}

export interface AcMeshMrDetail {
  mr: AcMeshMrStatus
  current_links: AcMeshLinkRecord[]
  recent_events: AcMeshMrEvent[]
}

export interface AcMeshSnapshot {
  id: number
  session_id: string
  controller_id: string
  controller_name: string
  site_id: string
  collected_at: string
  ac_time: string
  source_type: string
  source_reference: string
  data_status: MeshDataStatus
  age_seconds: number | null
  link_count: number
  parse_status: string
  error_summary: string
}

export interface AcMeshSnapshotPage {
  items: AcMeshSnapshot[]
  total: number
  page: number
  page_size: number
}

export interface AcMeshRawTail {
  snapshot_id: number | null
  available: boolean
  lines: string[]
  line_count: number
  source_reference: string
  updated_at: string
  message: string
}

export interface AcMeshMrQuery {
  online_status?: string
  train_no?: string
  mr_name?: string
  station?: string
  section?: string
  line_side?: string
  peer_ap_name?: string
  unmatched_only?: boolean
  query?: string
  page: number
  page_size: number
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}

export interface AcMeshLinkQuery {
  controller_id?: string
  mr_name?: string
  mr_mac?: string
  peer_ap_name?: string
  peer_ap_mac?: string
  station?: string
  section?: string
  line_side?: string
  match_status?: string
  query?: string
  page: number
  page_size: number
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}

File: src/netconsole/models/api/ac_mesh_link.py
from __future__ import annotations

from pydantic import Field

from netconsole.models.api.common import ApiModel


class AcMeshLinkRecordDTO(ApiModel):
    id: int
    snapshot_id: int
    controller_id: str = ""
    controller_name: str = ""
    mr_id: str
    train_no: str = ""
    car_end: str = ""
    mr_name: str = ""
    mr_mac: str = ""
    mr_device_id: str = ""
    mr_management_ip: str = ""
    mr_online_status: str = "unknown"
    peer_ap_id: str = ""
    peer_ap_name: str = ""
    peer_ap_mac: str = ""
    peer_radio: str = ""
    mesh_interface: str = ""
    rssi: int | None = None
    station: str = ""
    section: str = ""
    mileage: str = ""
    line_side: str = ""
    ap_rx_power: str = ""
    switch_rx_power: str = ""
    last_seen_at: str = ""
    match_method: str = "unmatched"
    match_warning: str = ""
    data_status: str = "no_data"


class AcMeshLinkPageDTO(ApiModel):
    items: list[AcMeshLinkRecordDTO] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 50


class AcMeshMrStatusDTO(ApiModel):
    mr_id: str
    train_no: str = ""
    train_display_name: str = ""
    car_end: str = ""
    mr_name: str = ""
    mr_mac: str = ""
    mr_device_id: str = ""
    management_ip: str = ""
    online_status: str = "unknown"
    peer_ap_id: str = ""
    peer_ap_name: str = ""
    peer_ap_mac: str = ""
    mesh_radio: str = ""
    rssi: int | None = None
    station: str = ""
    section: str = ""
    mileage: str = ""
    line_side: str = ""
    ap_rx_power: str = ""
    switch_rx_power: str = ""
    last_seen_at: str = ""
    match_method: str = "unmatched"
    match_warning: str = ""
    data_status: str = "no_data"


class AcMeshMrPageDTO(ApiModel):
    items: list[AcMeshMrStatusDTO] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 50


class AcMeshMrEventDTO(ApiModel):
    id: int
    event_time: str = ""
    event_type: str = ""
    status: str = ""
    station: str = ""
    ap_name: str = ""
    rssi: int | None = None
    car_end: str = ""


class AcMeshMrDetailDTO(ApiModel):
    mr: AcMeshMrStatusDTO
    current_links: list[AcMeshLinkRecordDTO] = Field(default_factory=list)
    recent_events: list[AcMeshMrEventDTO] = Field(default_factory=list)


class AcMeshSnapshotDTO(ApiModel):
    id: int
    session_id: str
    controller_id: str = ""
    controller_name: str = ""
    site_id: str
    collected_at: str = ""
    ac_time: str = ""
    source_type: str = "vehicle_mr_online_snapshot"
    source_reference: str = ""
    data_status: str = "no_data"
    age_seconds: int | None = None
    link_count: int = 0
    parse_status: str = ""
    error_summary: str = ""


class AcMeshSnapshotPageDTO(ApiModel):
    items: list[AcMeshSnapshotDTO] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 30


class AcMeshSnapshotDetailDTO(ApiModel):
    snapshot: AcMeshSnapshotDTO
    links: list[AcMeshLinkRecordDTO] = Field(default_factory=list)


class AcMeshLinkSummaryDTO(ApiModel):
    site_id: str
    controller_id: str = ""
    controller_name: str = ""
    registered_mrs: int = 0
    online_mrs: int = 0
    offline_mrs: int = 0
    stale_mrs: int = 0
    unknown_mrs: int = 0
    active_links: int = 0
    link_total: int = 0
    unmatched_links: int = 0
    offline_ap_links: int = 0
    updated_at: str = ""
    age_seconds: int | None = None
    data_status: str = "no_data"
    source_type: str = "vehicle_mr_online_snapshot"
    raw_available: bool = False
    message: str = ""


class AcMeshRawTailDTO(ApiModel):
    snapshot_id: int | None = None
    available: bool = False
    lines: list[str] = Field(default_factory=list)
    line_count: int = 0
    source_reference: str = ""
    updated_at: str = ""
    message: str = "暂无 Mesh-Link 原始数据"


class AcMeshLinkRefreshRequestDTO(ApiModel):
    controller_id: str = Field(min_length=1, max_length=100)
    include_switch_history: bool = False


class AcMeshLinkRefreshResponseDTO(ApiModel):
    success: bool = True
    task_id: str
    status: str
    already_running: bool = False
    message: str


__all__ = [
    "AcMeshLinkPageDTO",
    "AcMeshLinkRecordDTO",
    "AcMeshLinkRefreshRequestDTO",
    "AcMeshLinkRefreshResponseDTO",
    "AcMeshLinkSummaryDTO",
    "AcMeshMrDetailDTO",
    "AcMeshMrEventDTO",
    "AcMeshMrPageDTO",
    "AcMeshMrStatusDTO",
    "AcMeshRawTailDTO",
    "AcMeshSnapshotDTO",
    "AcMeshSnapshotDetailDTO",
    "AcMeshSnapshotPageDTO",
]
